packages/bot-web-ui/src/components/journal/journal.tsx

```diff
@@ -111,10 +127,10 @@ const FilterDialog = ({
     filterMessage,
     is_filter_dialog_visible,
     toggleFilterDialog,
-}) => {
-    const wrapper_ref = React.useRef();
-
-    const validateClickOutside = event => is_filter_dialog_visible && !toggle_ref.current.contains(event.target);
+}: TFilterDialogProps) => {
+    const wrapper_ref = React.useRef<HTMLDivElement>(null);
+    const validateClickOutside = (event: React.ChangeEvent<HTMLInputElement>) =>
+        is_filter_dialog_visible && toggle_ref.current && !toggle_ref.current.contains(event.target);
```

Comment:
shorter form: 
  is_filter_dialog_visible && !toggle_ref?.current.contains(event.target);

Reply:
@sara-fs in this short case we will get an error 'Object is possibly 'null'. Type script requires a check that the object `toggle_ref` exists.

Reply:
Optional chaning is allowed as of version 3.7 of typescript. It's not a problem anymore.